src/EditorFeatures/DiagnosticsTestUtilities/CodeActions/CSharpCodeRefactoringVerifier`1+Test.cs

```diff
@@ -85,8 +92,70 @@ public Test()
             public string? EditorConfig { get; set; }
 
 #if !CODE_STYLE
+            private readonly List<AdhocWorkspace> _workspaces = new();
+
             protected override AnalyzerOptions GetAnalyzerOptions(Project project)
                 => new WorkspaceAnalyzerOptions(base.GetAnalyzerOptions(project), project.Solution);
+
+            /// <summary>
+            /// The <see cref="TestHost"/> we want this test to run in.  Defaults to <see cref="TestHost.InProcess"/> if unspecified.
+            /// </summary>
+            public TestHost TestHost { get; set; } = TestHost.InProcess;
+
+            /// <summary>
+            /// The set of code action <see cref="CodeAction.Title"/>s offered the user in this exact order.
+            /// Set this to ensure that a very specific set of actions is offered (or not).
+            /// </summary>
+            public string[]? ExactActionSetOffered { get; set; }
+
+            protected override ImmutableArray<CodeAction> FilterCodeActions(ImmutableArray<CodeAction> actions)
+            {
+                var result = base.FilterCodeActions(actions);
+
+                if (ExactActionSetOffered != null)
+                {
+                    Verify.Equal(ExactActionSetOffered.Length, result.Length, "Different number of code actions provided");
```

Comment:
nit:

```suggestion
                    Verify.Equal(ExactActionSetOffered.Length, result.Length, $"Different number of code actions provided. Expected: {ExactActionSetOffered.Length}, Actual: {result.Length}.");
```

Reply:
i think that isn't really needed as that's what Verify.Equal should already indicate :)

Reply:
> i think that isn't really needed as that's what Verify.Equal should already indicate :)

Yeah. I think I should sleep :smile:

Reply:
💡 Use `SequenceEqual` with a comparer.

Reply:
will do.